client/cgo/verification.go

```diff
@@ -112,8 +142,15 @@ func cgoVerifyAuthPath(treeHash unsafe.Pointer, treeHashSize C.int,
 	ap.TreeNonce = tn
 	ap.LookupIndex = li
 	ap.PrunedTree = pt
+	ap.Leaf = &leafNode{
```

Comment:
Kind of doubled down on the `nil`s here, because doing this makes the signature to the verification functions a lot cleaner, and clearer.

And, the important point, is that I think we'll see that as we finish up the protocol, the client is going to be receiving messages that unmarshall like this anyways.